lib/diagnostic.rb

```diff
@@ -1,44 +1,53 @@
 # frozen_string_literal: true
-
+list = 1..10
 # Ruby enumerable diagnostic
 class Diagnostic
   # Question 1
   # Write ruby code to return the longest entry in the list supplied.
   # Assume each entry in the list implements a length method that returns
   # a Fixnum.
   def longest(list)
-    # Your code goes here
+    list.max
   end
-
+  longest(list)
   # Question 2
   # Write ruby code to return the sum of entries in the
   # list supplied (i.e. apply the `+` operator).
   # Assume each entry implements a `+` method.
   def sum(list)
-    # Your code goes here
+    list.reduce(:+)
   end
+  sum(list)
 
   # Question 3
   # Write ruby code that returns true if some of the entries
   # in the list of numbers are odd
-  def some_odd(numbers)
-    # Your code goes here
+  def some_odd(list)
+    list.any? { :odd? }
   end
+  some_odd(list)
 
   # Question 4
   # Write ruby code that returns true if every entry in the
   # list of numbers are even
   def every_even(numbers)
-    # Your code goes here
+    # numbers.select { :even? } NOT SURE WHY THIS ISN:T WORKING BUT WHEN I TEST IT BOMBS
+    numbers.select { |num| num.even? }
```

Comment:
Again you are using braces and making a block but you should just use parens:

`numbers.all?(&:even?)`